js/multiFileUpload.js

```diff
@@ -1,351 +1,390 @@
-          {
-            var fileName = null;
-            index = String(index);
+          return false;
+        }
+      }
+      else {
+        // Upload to AtoM - wait on promise until all complete.
+        this.uppy.upload().then((result) => {
+          if (result.failed.length > 0) {
+            (this.checkUploadSuccessful() === true) ?
+              this.showAlert(Qubit.multiFileUpload.i18nSomeFilesFailedError, 'alert-error') :
+              this.showAlert(Qubit.multiFileUpload.i18nNoSuccessfulFilesError, 'alert-error');
+
+            this.$retryButton.show();
+          }
+          else {
+            this.$submitButton.attr('disabled', 'disabled');
+            this.$cancelButton.removeAttr("href").attr('disabled', 'disabled');
+            this.showAlert(Qubit.multiFileUpload.i18nImporting, 'alert-info');
+            // Post to multiFileUpload.
+            $('#multiFileUploadForm').submit();
+          }
+        })
+      }
+
+      return false;
+    },
+
+    // Push a record of successful file upload into array uploadItems. 
+    // These will be added to this array in order of when they completed uploading.
+    // This info is needed to build the hidden form elements once all files 
+    // have completed uploading to AtoM.
+    onUploadSuccess: function (file, response)
+    {
+      this.uploadItems.push({file, response});
+    },
+
+    // onComplete runs when all uploads are complete - even if there were errors.
+    // Adds the form elements in the same order as result.successful so that
+    // they are imported into AtoM: Image 01, Image 02, etc.
+    onComplete: function (result)
+    {
+      // Iterates over successfully uploaded items.
+      var uploadItems = this.uploadItems;
+
+      $.each(result.successful, function(key, file) {
+        // Get the corresponding upload response.
+        var fileResponse = uploadItems.find(x => x.file.id === file.id).response;
+
+        // Add hidden form elements for each successfully uploaded file.
+        $('<div class="multiFileUploadItem" id=' + file.id + '>' +
+          '<div class="multiFileUploadInfo">' +
+            '<div class="form-item">' +
+              '<input type="hidden" class="filename" value="' + fileResponse.body.name + '"/>' +
+              '<input type="hidden" class="md5sum" value="' + fileResponse.body.md5sum + '"/>' +
+              '<input type="hidden" name="files[' + file.id + '][name]" value="' + fileResponse.body.name + '"/>' +
+              '<input type="hidden" name="files[' + file.id + '][md5sum]" value="' + fileResponse.body.md5sum + '"/>' +
+              '<input type="hidden" name="files[' + file.id + '][tmpName]" value="' + fileResponse.body.tmpName + '"/>' +
+              '<input type="hidden" class="title" name="files[' + file.id + '][infoObjectTitle]" value="' + file.meta.title + '"/>' +
+            '</div>' +
+          '</div>' +
+        '</div>')
+        .appendTo("#uploads");
+      });
```

Comment:
I know it's not the best at this stage, but if it's possible, I'd rather see the slugs being used all over the place instead of DB ids.

Reply:
These are not object.id values but rather Uppy's unique id's for each image. At this point in the import, these digital objects do not yet have a slug or id.

Reply:
Oh, I see!